src/app/app.component.html

```diff
@@ -215,28 +247,47 @@ <h2 class="text-center mt-5 text-muted">No messages</h2>
               <option value="" disabled selected>Recipient</option>
               <option *ngFor="let guest of activeMailbox.recipients" value="{{ guest }}">{{ guest }}</option>
             </select>
-            <textarea ngModel name="message" placeholder="Message" rows="3" required class="form-control"></textarea>
+            <textarea ngModel name="message" #message placeholder="Message"
+              rows="3" required class="form-control"></textarea>
             <button class="btn btn-success" type="submit" [disabled]="!messageForm.valid">Send</button>
           </form>
         </div>
       </div>
 
+      <div class="row" *ngIf="viewMailboxSubscreen === 'sendFile'">
+        <div class="col-sm-6">
+          <span class="badge bg-success" [class.badge-show]="UIFlags[UIAction.fileSent]">File sent!</span>
+          <form class="d-grid gap-2 needs-validation" #fileForm="ngForm" novalidate
+            [class.was-validated]="fileForm.submitted && fileForm.touched"
+            (submit)="sendFile(fileRecipient.value); fileForm.reset()">
+            <select type="text" name="fileRecipient" #fileRecipient ngModel class="form-select" required>
+              <option value="" disabled selected>Recipient</option>
+              <option *ngFor="let guest of activeMailbox.recipients" value="{{ guest }}">{{ guest }}</option>
+            </select>
+            <input type="file" name="file" #file ngModel class="form-control" required
+              (change)="selectedFile = $event.target.files.item(0)">
+            <button class="btn btn-success" type="submit" [disabled]="!fileForm.valid">Send</button>
+          </form>
+        </div>
+      </div>
+
       <div class="row" *ngIf="viewMailboxSubscreen === 'addKey'">
         <div class="col-sm-6">
-          <span class="badge bg-success" [ngClass]="{'badge-show': keyAdded}">Key added!</span>
+          <span class="badge bg-success" [class.badge-show]="UIFlags[UIAction.keyAdded]">Key added!</span>
           <form class="d-grid gap-2 needs-validation" #keyForm="ngForm" novalidate
             [class.was-validated]="keyForm.submitted && keyForm.touched"
-            (submit)="addPublicKey(activeMailbox, keyForm)">
-            <input type="text" ngModel name="name" placeholder="User Name" class="form-control" required>
-            <input type="text" ngModel name="key" placeholder="Public Key (Base64)" class="form-control" required>
+            (submit)="addPublicKey(keyName.value, keyKey.value); keyForm.reset()">
+            <input type="text" ngModel name="name" #keyName placeholder="User Name" class="form-control" required>
+            <input type="text" ngModel name="key" #keyKey placeholder="Public Key (Base64)"
+              class="form-control" required>
             <button class="btn btn-success" type="submit" [disabled]="!keyForm.valid">Add</button>
           </form>
         </div>
```

Comment:
also it would be nice to move some part of app to separate component. For example relay, adding mailbox, and mailbox control

Reply:
Hm, I'm not sure. I agree that `GlowService` makes sense, it'll be easy to read all Glow-related quickly separated from UI. And the component will become much shorter. But splitting the component of a micro app into 20-line subcomponents will only make it harder to follow. After all, we don't need to scale this app, components will never be used elsewhere. Your thoughts?

Reply:
it works for me even without glow service. I thought this code will be for public access.